### PR TITLE
Move from deprecated Composer 1 to Composer 2 runtime api

### PR DESCRIPTION
**Pull request by DeyV**

### CODE REVIEW

**Comment by orklah** [comment]
phar build is failing in CI.

It must have something to do with `scoper.inc.php` file, but I was unable to see why PackageVersion worked but not the new package. Can you try to tweak with that to see if you can fix it?

**Comment by DeyV** [comment]
I fixed this problem. 

To full remove dependency of composer/package-versions-deprecated 
release of https://github.com/psalm/psalm-plugin-phpunit/pull/116 is needed

File: src/Psalm/Internal/CliUtils.php [hunk 1]
```diff
@@ -147,8 +147,13 @@ public static function requireAutoloaders(
             exit(1);
         }
 
-        define('PSALM_VERSION', Versions::getVersion('vimeo/psalm'));
-        define('PHP_PARSER_VERSION', Versions::getVersion('nikic/php-parser'));
+        try {
```
Comment (orklah) [inline src/Psalm/Internal/CliUtils.php:150]: Please add $version = null; just above the try. Psalm does not like working with possibly undefined variables

**Comment by orklah** [comment]
Thanks!

**Comment by weirdan** [comment]
I think this broke psalm.dev:

```
Psalm runner output: 

Internal Psalm error on line 22 - /src/OnlineChecker.php: Class "PackageVersions\Versions" not found
```

**Comment by weirdan** [comment]
Fixed in https://github.com/psalm/psalm.dev/commit/8c65f63cdeb9c08102067ea398061686237704e7

**Pull request merged**